### ISSUE TITLE
Serve other static `index.html` and assets on subpath(s)?

### ISSUE THREAD

**Issue by chris-dura** [vercel/serve#757]
### Description

I basically have 3 "apps", each are "static" bundles with their own relative assets.

```
my-app/
├─ app2/
│  ├─ app2.js
│  ├─ index.html
├─ app3/
│  ├─ index.css
│  ├─ app3.js
├─ index.html
├─ my-app.js
```

When I run `serve -l 5280 my-app`, I can browse to `http://localhost:5280` and it serves `my-app/index.html` and the `my-app.js` just fine.

However, when I type `http://localhost:5280/app2` in the address bar, the `app2/index.html` loads, but I get a 404 on `app2.js` :/

### Library version

v14.2.0

### Node version

v16.17.0

**Comment by aasen** [comment]
I have the same issue.

**Comment by yairEO** [comment]
Even if you will directly run `serve app2/index.html` you'll get `404` for the assets (but the HTML itself will be served...)

**Comment by chris-dura** [comment]
> Even if you will directly run `serve app2/index.html` you'll get `404` for the assets (but the HTML itself will be served...)

... do you mean `serve my-app/app2`? Because, it should get the assets just fine in that case, since `app2/index.html` has _relative_ asset URLs... eg, `<script src="./app2.js"/>`

**Comment by yairEO** [comment]
No, I mean, if you directly serve an `.html` file, all its assets return `404`

**Comment by yairEO** [comment]
@chris-dura - I do not want to serve **folders** in any way. I **only** need to serve single files:

```
serve my-app/path/to/some/1.html
serve my-app/path/to/some/2.html
serve my-app/path/to/some/3.html
```

and so on

**Comment by chris-dura** [comment]
@yairEO -- Right, there's nothing in the docs that indicate serving single files like that would be supported.

I understand if there's a reason you can't do this, but if you change them to `index.html` files...

```
my-app/path/to/some/1/index.html
my-app/path/to/some/2/index.html
my-app/path/to/some/3/index.html
```
... then this would work out-of-the-box with no extra configuration ...

```
serve my-app/path/to/some/1
serve my-app/path/to/some/2
serve my-app/path/to/some/3
```

You might also be able to configure some [redirects/rewrites](https://github.com/vercel/serve-handler#rewrites-array) in `serve.json`.

**Comment by yairEO** [comment]
I cannot change them to `index.html` because they are all in the same folder and cannot be moved. Also, the name of the `html` files is important.

I am using [`http-server`](https://www.npmjs.com/package/http-server) which can do this extremely simple thing, which `serve` cannot.

**Comment by ultravires** [comment]
I have the same issue.

**Comment by pandanoir** [comment]
Adding singleFile option and using it to rewrite destination will resolve this issue.

```js
   config.rewrites = [ 
     { 
       source: '**', 
       destination: args['--singleFile'],
     }, 
     ...existingRewrites, 
   ]; 
```

(The above code is pseudocode, so it might not work)

https://github.com/vercel/serve/blob/c8b7436b8b6e8db5279357638a6556ea7a562a6c/source/main.ts#L78-L90

**Comment by pandanoir** [comment]
Temporary solution: Write a `serve.json` and set the `--config` option

`serve build --config ../serve.json`

directory structure:
```
packages/internal-cms
├── build
│  └── path/to/index.html
└── serve.json
```

serve.json:

```
{
  "rewrites": [
    {
      "source": "**",
      "destination": "/path/to/index.html"
    }
  ]
}
```

**Comment by nbilyk** [comment]
same. Switched to http-server which doesn't have this issue.
